rearrange_snp_matrix.pl

```diff
@@ -0,0 +1,326 @@
+	
+	my $internalNumber = 1;
+	foreach my $node ( $input_taxa_tree->get_nodes ){
+	  my $nodeBranchLength = $node->branch_length();
+	  $nodeBranchLength = 0 if !defined $nodeBranchLength;
+	  my $lengthToSNP = $nodeBranchLength*$treeTotalSNP;
+	  $lengthToSNP = sprintf "%.2f", $lengthToSNP;
+	  my $nodeName = $node->id();
+      
+      if($lengthToSNP<0.5){
+      	$node->branch_length(0);
+      }
+      else{
+      	#round the number printed to the closest integer value
+      	$node->branch_length(round($lengthToSNP));
+      }
+      $internalNumber++ if !$node->is_Leaf;
+    }
+    close($inputPhy);
+    print $logger "Branch lengths have been successfully converted to total SNP's\n";
+}
+
+#==============================================================================
+#Purpose:
+#	Exponentially scales branch lengths on the input tree to allow the user to 
+#	make the final output more easily human readable.
+#Input:
+#	$input_taxa_tree -> Bio::Phylo::Forest::Tree phylogenetic tree to re-root
+#   $exponent -> the exponent to factor all branch lengths by
+#	$logger -> Reference to a Logger object.
+#Output:
+#	matrix.csv file with resized branch lengths.
+#==============================================================================
+sub resizeTree
+{
+	my ($input_taxa_tree, $exponent, $logger) = @_;
+	
+	my $minimumBranchLength = 0.009;
+	#zero out the branch values that are lower than threshold to maintain tree structure for near 0 branch lengths:
+	foreach( $input_taxa_tree->get_nodes){
+		if($_->get_branch_length() < $minimumBranchLength){
+			$_->set_branch_length(0);
+		}
+	}
+	
+	$input_taxa_tree->exponentiate($exponent);
+	print $logger "Branch lengths have been resized with an exponential factor of: ".$exponent."\n";
+}
+
+#==============================================================================
+#Purpose:
+#	Script to re-root a phylogenetic tree, order the tree in increasing/
+#	decreasing order, convert branch lengths to total SNP's, and output a 
+#	revised matrix.csv to match the phylogenetic tree.
+#==============================================================================
+
+my ($tmp_dir, $keep_tmp, $convert, $root_strain, $tree_order, $input_tree, $output_dir, $matrix_input, $input_phy, $help );
+
+GetOptions(
+	't|tmp-dir=s' => \$tmp_dir,
+	'k|keep_tmp=s' => \$keep_tmp,
+	'c|convert' => \$convert,
+	'r|root=s' => \$root_strain,
+	's|sort=s' => \$tree_order,
+	'i|inputtree=s' => \$input_tree,
+	'o|outdir=s' => \$output_dir,
+	'm|matrix=s' => \$matrix_input,
+	'p|phy=s' =>	\$input_phy,
+	'h|help' => \$help
+);
+#handle documentation and help
+if ($help){
+	pod2usage(	-verbose => 99,
+				-sections => "SYNOPSIS|OPTIONS|DESCRIPTION");
+}
+
+pod2usage(1) unless $tmp_dir && $input_tree && $output_dir && $matrix_input && $input_phy;
+
+#check to ensure all required command line variables are present and set default values if applicable:
+die "Error: No temp directory defined." if (not defined $tmp_dir);
```

Comment:
maybe not die if no $tmp_dir was given. Otherwise we cannot easily put this on Galaxy. You could have a default to use the current directory of where the user is running the script.  When Galaxy runs a job, it creates it's own temp directory and runs the script from there. So the default can be '.'